cmake/BundledAzureSdkCpp.cmake

```diff
@@ -55,14 +59,17 @@ function(use_bundled_libazure SOURCE_DIR BINARY_DIR)
 
     set(AZURE_SDK_CMAKE_ARGS ${PASSTHROUGH_CMAKE_ARGS}
         -DWARNINGS_AS_ERRORS=OFF
-        -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR})
+        -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+        -DDISABLE_AZURE_CORE_OPENTELEMETRY=ON
+        -DBUILD_TRANSPORT_CURL=ON
+        -DCMAKE_CXX_FLAGS=-DCURL_STATICLIB)
```

Comment:
Would it work if we used `target_compile_definitions` instead, or does that not get passed down? Also, do we not need this definition when including Azure SDK headers?

Reply:
moved this directive to the curl target